Course Discussion MVVM

varunasingh opened the pull request.

varunasingh committed:
Separate branch for course discussion detail

mikedawson commented on core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/CourseDiscussionDetailViewModel.kt:
```diff
@@ -0,0 +1,10 @@
+import com.ustadmobile.lib.db.entities.DiscussionPostWithDetails
+
+data class CourseDiscussionDetailUiState(
+    val courseDiscussion: CourseDiscussion? = null,
```
I think as per call we don't need CourseDiscussion entity. Title, description, etc are on the courseblock.

mikedawson commented on app-android/src/main/java/com/ustadmobile/port/android/view/CourseDiscussionDetailFragment.kt:
```diff
@@ -145,6 +185,124 @@ class CourseDiscussionDetailFragment: UstadDetailFragment<CourseDiscussion>(),
+        ){ post ->
+
+            val context = LocalContext.current
+            val datePosted = remember { DateFormat.getDateFormat(context)
```
Please use rememberFormattedDate. This doesn't have a key set, so we haven't told compose what would invalidate it. There needs to be an explicit invalidation key, e.g. the timestamp. This is done by the rememberformatteddate function.

mikedawson commented on app-android/src/main/java/com/ustadmobile/port/android/view/CourseDiscussionDetailFragment.kt:
```diff
@@ -145,6 +185,124 @@ class CourseDiscussionDetailFragment: UstadDetailFragment<CourseDiscussion>(),
+                },
+                secondaryText = {
+                    Column {
+                        Text(post.discussionPostTitle?: "")
```
Shouldn't the title be in the primary text, then the latest message is in the secondary text? I don't think we need the post message itself.

mikedawson commented on app-android/src/main/java/com/ustadmobile/port/android/view/composable/UstadMessageField.kt:
```diff
@@ -0,0 +1,201 @@
+ * Main field layout for message fields in DetailViews
+ */
+@Composable
+fun UstadMessageField(
```
This is not used, so it should be removed.

mikedawson commented on app-react/src/main/kotlin/com/ustadmobile/view/CourseDiscussionDetailScreen.kt:
```diff
@@ -0,0 +1,173 @@
+                            Date(item.postLatestMessageTimestamp ?: 0L).toLocaleDateString()
+                        }
+
+                        UstadMessageField{
```
As per discussion, we should be using the standard ListItem and its formatting.

mikedawson commented on app-react/src/main/kotlin/com/ustadmobile/mui/components/UstadMessageField.kt:
```diff
@@ -0,0 +1,266 @@
+import react.ReactNode
+import react.create
+
+external interface UstadMessageFieldProps: Props {
```
This should be removed as we should be using the standard ListItem

mikedawson reviewed (changes requested):
Thanks, please see the comments. Please remove the use of UstadMessageField and use the standard ListItem on React/MUI. Please remove  the (unused) UstadMessageField on Jetpack compose.

varunasingh committed:
Updated code with comments and feedback.

varunasingh committed:
Merge branch 'dev-mvvm-primary' into dev-mvvm-discussiondetail

varunasingh committed:
Update primary to post detail

varunasingh committed:
Added fragments as well

varunasingh committed:
separate branch for post edit

varunasingh committed:
Merge branch 'dev-mvvm-primary' into dev-mvvm-discussionpostedit

varunasingh committed:
Merge branch 'dev-mvvm-primary' into dev-mvvm-discussionpostdetail

varunasingh committed:
Merge branch 'dev-mvvm-primary' into dev-mvvm-discussiondetail

varunasingh committed:
Adding models and tests

varunasingh committed:
Merge branch 'dev-mvvm-primary' into dev-mvvm-discussionpostdetail

varunasingh committed:
Merge branch 'dev-mvvm-primary' into dev-mvvm-discussionpostdetail

varunasingh committed:
New UI and View model

varunasingh committed:
Android preview OK

varunasingh committed:
Add reply button ok

varunasingh committed:
Merge branch 'dev-mvvm-primary' into dev-mvvm-discussionpostdetail

varunasingh committed:
merge fixes

varunasingh committed:
Minor clean up

varunasingh committed:
Merge branch 'dev-mvvm-discussiondetail' into dev-mvvm-discussionpostedit

varunasingh committed:
Merge branch 'dev-mvvm-discussionpostdetail' into dev-mvvm-discussionpostedit
Updates to viewModel and on mvvm and jetpack

varunasingh committed:
Merge branch 'dev-mvvm-primary' into dev-mvvm-discussionpostdetail

varunasingh committed:
Fix

varunasingh committed:
Discusion detail updates.

varunasingh committed:
Updates

varunasingh committed:
Persisting

varunasingh committed:
Updates before merge

varunasingh committed:
Merge branch 'dev-mvvm-primary' into dev-mvvm-discussiondetail

varunasingh committed:
Fix alignment issue

varunasingh committed:
Replies working

varunasingh committed:
Fix reply list not working

varunasingh committed:
Merge branch 'dev-mvvm-primary' into dev-mvvm-discussiondetail

varunasingh committed:
Updates to html

varunasingh committed:
Merge branch 'dev-mvvm-primary' into dev-mvvm-coursediscussiondetail

varunasingh committed:
Merge branch 'dev-mvvm-coursediscussiondetail' into dev-mvvm-discussiondetail

mikedawson committed:
Fix JS-Joda timezone usage on JS/production. This was being optimized out of the DCE'd version.

mikedawson committed:
Fix creation of server distribution zip. No java module info required.

mikedawson committed:
Update runserver.sh and documentation to reflect switch to using Ktor plugin instead of shadowJar.

mikedawson committed:
Update server distribution build to include and use ustad-server.conf . Use custom script templates to add arguments for Unix and Windows.

mikedawson committed:
Update server distribution version to put data directory and log directory in the expected dirs when running from distribution zip.

mikedawson committed:
Fix server distribution zip to run as expected on Windows. Update classpath to avoid windows command line limit issue.

mikedawson committed:
Add implementation of RedirectViewModel and component for React (bring user to correct screen on first page opening). Setup react router index to use redirect component.

Fix NavHost to handle first location key correctly: avoid going back when there is nothing left on the back stack.

Fix handling detection of apiUrl on Javascript when running in production.

mikedawson committed:
Delete obsolete fragment test (being replaced in MVVM).

mikedawson committed:
Fix sharedse Android compilation issues.

mikedawson committed:
Update INSTALL documentation to reflect the new server packaging.

mikedawson committed:
Fix generation of Windows startup script: using set with "*" appeared to confuse the template engine.

mikedawson committed:
Add white background for diagrams

varunasingh committed:
Merge branch 'dev-mvvm-primary' into dev-mvvm-discussiondetail

varunasingh committed:
Fragments working OK

mikedawson commented:
Already updated/ merged / updated in dev-mvvm-primary
